flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java

```diff
@@ -157,8 +159,12 @@ public static JobManagerSharedServices fromConfiguration(
 			throw new IllegalConfigurationException(AkkaUtils.formatDurationParsingErrorMessage());
 		}
 
+		final int poolSize = config.getInteger(ClusterOptions.CLUSTER_IO_EXECUTOR_POOL_SIZE, Hardware.getNumberCPUCores());
+		Preconditions.checkArgument(poolSize > 0,
+			String.format("Illegal pool size (%s) of io-executor for cluster entry-point, please re-configure '%s'.",
+				poolSize, ClusterOptions.CLUSTER_IO_EXECUTOR_POOL_SIZE.key()));
 		final ScheduledExecutorService futureExecutor = Executors.newScheduledThreadPool(
-				Hardware.getNumberCPUCores(),
+				poolSize,
 				new ExecutorThreadFactory("jobmanager-future"));
```

Comment:
Instead of making the future executor of the `JobManager` configurable I would suggest to make the `ioExecutor` of the `ClusterEntrypoint` configurable: https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java#L263

Reply:
It's my fault, will fix this.